multiraft/multiraft.go

```diff
@@ -568,8 +625,29 @@ func (s *state) start() {
 					s.coalescedHeartbeat()
 				}
 
-			case cb := <-s.callbackChan:
-				cb()
+			case eventsChan <- event:
+				// Send as many as possible events to eventsChan
```

Comment:
Why do you have this extra loop? You could just set `s.pendingEvents = s.pendingEvents[1:]` and let the next event be sent on the next loop iteration.